### PR TITLE
Update Docs

### PR DESCRIPTION
**Pull request by cgiroux86**
Update docs with information regarding forced revisions and prior actions.

---
This change is [Reviewable](https://reviewable.io/reviews/reviewable/reviewable/828)

### CODE REVIEW

**Review by cgiroux86 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/reviewable/reviewable/828)* status: 0 of 5 files reviewed, all discussions resolved (waiting on @pkaminski)

*[a discussion](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6QbRABivg3iVhFPeh:-MY6QbRBDldkTsKR0lmq:b-z2dmdn) (no related file):*
BTW, Headway is updated and could use a proofread if you wouldn't mind :) 

---

File: enterprise/changelog.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ New releases are announced on the [reviewable-enterprise-announce mailing list](
 - Fix: adapt to new GitHub OAuth token format.  This fixes "Unable to decrypt token with any key" errors.  You do _not_ need to change or fix your token encryption private key.
 - Fix: reduce Docker image size back <90MB.  The previous release accidentally bloated it a bit.
 - Fix: ensure that critical file diffs are proposed, even if the prior revision was modified to base. [See issue #342](https://github.com/Reviewable/Reviewable/issues/342)
+- New: display a revision cell with both an `action` and a `prior action` when rebased revisions are matched using heuristics. See the [docs](https://docs.reviewable.io/files.html#revision-cells) for details.
```
Comment (pkaminski) [inline enterprise/changelog.md:15]: I'd say this is more of an `Upd`.  Also, note that within each version I sort entries as `New` -> `Upd` -> `Fix`. 

I'd reword this to be more end-user-friendly, since `action` and `prior action` are purely internal terms:  `display an inner status color in rebased revision cells relative to the matched revision, which might not be the preceding one.  See the [docs]() for details.`

Should also have a separate entry about the reverted marker.

File: docs/files.md [hunk 2]
```diff
@@ -2,16 +2,25 @@
...
 
 The brackets in a row are the current left and right diff bounds for the file. To adjust the diff bounds, click on one desired revision bound and drag to the other one.  You can also just click on a revision to adjust the nearest bound.
 
 ![reviewable file matrix diff bounds](images/filematrix_8.png)
 
+By design, Reviewable doesn't track files that haven't deviated from their base form -- including files that have been modified back to their original state.  However, under certain circumstances (e.g., [rebasing](#rebasing)), Reviewable will "force" a final revision when it otherwise wouldn't.  This is indicated by a reverted icon in a kept cell and is designed to prevent you from missing a critical diff.  Reviewable will also add special styling to cells when [rebased heuristics](#heuristics) are in play.
+
+Symbol | Meaning
+:----: |---------
+![reviewable forced revision](images/filematrix_12.png) | File revision forced.
```
Comment (pkaminski) [inline docs/files.md:20]: No need to call this out separately, having the entry in the file matrix status image above is sufficient.

File: docs/files.md [hunk 3]
```diff
@@ -2,16 +2,25 @@
...
+![reviewable prior action](images/filematrix_11.png) | <strong>Inner Disc Color</strong>: File action against matched prior rebased revision. <br></br> <strong>Outer Background Color</strong>: File action against immediately preceding revision. <br></br> In this example, <i>r6</i> is not modified compared to <i>r4</i>, but is modified compared to <i>r5</i>.
```
Comment (pkaminski) [inline docs/files.md:21]: The image should include the revision arc above -- it's currently cropped out.  Otherwise, there's no context for why we'd care about comparing `r6` to `r4`.

Also, `br` is an odd element.  It's technically self-closing so you just need to use `<br>` by itself, but I typically use the XHTML variant `<br/`> instead as it's both compact and doesn't trigger my unbalanced tags reflex.

File: docs/files.md [hunk 4]
```diff
@@ -2,16 +2,25 @@
...
 
-![reviewable file matrix status](images/filematrix_9.png)
+![reviewable file matrix status](images/filematrix_10.png)
 
 The brackets in a row are the current left and right diff bounds for the file. To adjust the diff bounds, click on one desired revision bound and drag to the other one.  You can also just click on a revision to adjust the nearest bound.
 
 ![reviewable file matrix diff bounds](images/filematrix_8.png)
 
+By design, Reviewable doesn't track files that haven't deviated from their base form -- including files that have been modified back to their original state.  However, under certain circumstances (e.g., [rebasing](#rebasing)), Reviewable will "force" a final revision when it otherwise wouldn't.  This is indicated by a reverted icon in a kept cell and is designed to prevent you from missing a critical diff.  Reviewable will also add special styling to cells when [rebased heuristics](#heuristics) are in play.
```
Comment (pkaminski) [inline docs/files.md:16]: This should all go into the rebasing section, since in fact that's the only circumstance where we'll need to force a revision.  Also, it's not entirely accurate:  we might need to force more than one revision (in case of multi-commit rebases), and the reverted icon isn't specific to forced revisions (any more).

**Review by pkaminski — CHANGES_REQUESTED** [review]
Doing this review in GitHub to get a feeling for just how much it sucks.  :)

**Review by cgiroux86 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/reviewable/reviewable/828)* status: 0 of 6 files reviewed, 6 unresolved discussions (waiting on @pkaminski)

___
*[docs/files.md, line 5 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS4-r2-5:-MYAv41S2SQDAfp8VoEU:b-896fix) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/docs/files.md#L5)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

The anchors are generated automatically when publishing based on the header text.  The explicit anchor below was only needed to provide backwards compatibility with links that used an older id.
</blockquote></details>

Done.
___
*[docs/files.md, line 10 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS4-r2-10:-MYAv3Cx0Be5U0VqpSGT:b-896fix) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/docs/files.md#L10)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

You should delete `filematrix_9.png` since we're not using it any more.  (Or just replace it with your updated screenshot.)
</blockquote></details>

Done.
___
*[docs/files.md, line 16 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS4-r2-16:-MYAv2WO8uOrwsX_UMy5:b-896fix) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/docs/files.md#L16)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

This should all go into the rebasing section, since in fact that's the only circumstance where we'll need to force a revision.  Also, it's not entirely accurate:  we might need to force more than one revision (in case of multi-commit rebases), and the reverted icon isn't specific to forced revisions (any more).
</blockquote></details>

Done.
___
*[docs/files.md, line 20 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS4-r2-20:-MYAv16r2GXPNA5AdegU:b-896fix) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/docs/files.md#L20)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

No need to call this out separately, having the entry in the file matrix status image above is sufficient.
</blockquote></details>

Done.
___
*[docs/files.md, line 21 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS4-r2-21:-MYAv0nf8KnCqIZnC68p:b-896fix) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/docs/files.md#L21)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

The image should include the revision arc above -- it's currently cropped out.  Otherwise, there's no context for why we'd care about comparing `r6` to `r4`.

Also, `br` is an odd element.  It's technically self-closing so you just need to use `<br>` by itself, but I typically use the XHTML variant `<br/`> instead as it's both compact and doesn't trigger my unbalanced tags reflex.
</blockquote></details>

Done.
___
*[enterprise/changelog.md, line 15 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS8-r2-15:-MYAzmSr7dz3DFtZ2xCv:b-896fix) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/enterprise/changelog.md#L15)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

I'd say this is more of an `Upd`.  Also, note that within each version I sort entries as `New` -> `Upd` -> `Fix`. 

I'd reword this to be more end-user-friendly, since `action` and `prior action` are purely internal terms:  `display an inner status color in rebased revision cells relative to the matched revision, which might not be the preceding one.  See the [docs]() for details.`

Should also have a separate entry about the reverted marker.
</blockquote></details>

Done.

**Review by cgiroux86 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/reviewable/reviewable/828)* status: 0 of 6 files reviewed, 7 unresolved discussions (waiting on @pkaminski)

*[a discussion](https://reviewable.io/reviews/reviewable/reviewable/828#-MYB0Y5gBWv0VBrn9Yvs:-MYB0Y5gBWv0VBrn9Yvt:b-nr4ck9) (no related file):*
Well, I got a merge conflict and tried to rebase with `master`, but now it's screwed up.

---

**Review by pkaminski — CHANGES_REQUESTED** [review]
> Reviewed 1 of 5 files at r1, 5 of 5 files at r3.
> *[Reviewable](https://reviewable.io/reviews/reviewable/reviewable/828)* status: all files reviewed, 6 unresolved discussions (waiting on @cgiroux86)

*[a discussion](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6QbRABivg3iVhFPeh:-MYBDEw5Aa4MbZv33hzG:b4y71rm) (no related file):*
<details><summary><i>Previously, cgiroux86 (Chris Giroux) wrote…</i></summary><blockquote>

BTW, Headway is updated and could use a proofread if you wouldn't mind :) 
</blockquote></details>

Can do, but please post in the Work channel so I can pin it to my inbox.  Otherwise there's approximately zero chance I'll remember to actually do it.  :)

---
*[a discussion](https://reviewable.io/reviews/reviewable/reviewable/828#-MYB0Y5gBWv0VBrn9Yvs:-MYBD51CAE2lc2BgliUE:buve00m) (no related file):*
<details><summary><i>Previously, cgiroux86 (Chris Giroux) wrote…</i></summary><blockquote>

Well, I got a merge conflict and tried to rebase with `master`, but now it's screwed up.
</blockquote></details>

Hmm, indeed, I didn't expect to see Anthony's commit in your PR.  Wanna try to figure it out?  If you get stuck, I'll see if I can stage a rescue tomorrow.  :)

---
___
*[docs/files.md, line 16 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS4-r2-16:-MYBDucT7UmKdgxeGxND:b-d17iv9) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/docs/files.md#L16)):*
<details><summary><i>Previously, cgiroux86 (Chris Giroux) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I don't think you addressed the accuracy part of the comment?
___
*[docs/files.md, line 35 at r3](https://reviewable.io/reviews/reviewable/reviewable/828#-MYBDXnh3QAHgGEOAaJd:-MYBDXnh3QAHgGEOAaJe:b-xhmhrr) ([raw file](https://github.com/reviewable/reviewable/blob/587f5d07d7b658e3fe503514514e05eb3e311baf/docs/files.md#L35)):*
> ```Markdown
> Review marks remain in a draft state and are only visible to you until [published](reviews.md#publish).  Recissions are publicized immediately however.
> 
> <a id="file-matrix"></a>
> ```

This anchor is also unnecessary.
___
*[docs/files.md, line 243 at r3](https://reviewable.io/reviews/reviewable/reviewable/828#-MYBDiSjCbSG7n2B3Zk5:-MYBDiSjCbSG7n2B3Zk6:bxuwpns) ([raw file](https://github.com/reviewable/reviewable/blob/587f5d07d7b658e3fe503514514e05eb3e311baf/docs/files.md#L243)):*
> ```Markdown
> Reviewable also uses heuristics to match rebased commits to their ancestors using commit messages.  When successful, this enables Reviewable to suggest minimum-delta revision pairs for diffing in the “review each commit” style even if you reordered the commits.  Arcs that link blocks of revisions above the file matrix show a high-level overview of this mapping.  (We don’t show matches between individual revisions because that can quickly lead to an unreadable diagram.) 
> 
> By design, Reviewable doesn't track files that haven't deviated from their base form -- including files that have been modified back to their original state.  However, when using heuristics, Reviewable will sometimes "force" a revision when it otherwise wouldn't.  This is indicated by a reverted icon in a kept cell and is designed to prevent you from missing a critical diff.  When applicable, revision cells in both the [file matrix](#file-matrix) and [file header](#file-header) will also have a unique styling. 
> ```

Typographical nit:  use `&mdash;` instead of `--`.  Also note the trailing space that's visible in the diff!
___
*[docs/images/filematrix_12.png, line 0 at r3](https://reviewable.io/reviews/reviewable/reviewable/828#-MYBERoxEFL0br9XZ_gy:-MYBERoy0oGMoPHobp7P:bohw9oi) ([raw file](https://github.com/reviewable/reviewable/blob/587f5d07d7b658e3fe503514514e05eb3e311baf/docs/images/filematrix_12.png#L0)):*
File no longer used, revert.
___
*[enterprise/changelog.md, line 15 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS8-r2-15:-MYBEl1qEO56RiNt4Pmy:b3ngd4v) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/enterprise/changelog.md#L15)):*
<details><summary><i>Previously, cgiroux86 (Chris Giroux) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

...and that entry about the reverted marker is also just an `upd`.  :)  Basically, if it just adds on to an existing feature (e.g., the file matrix) I treat it as an update unless it's really big.

**Review by cgiroux86 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/reviewable/reviewable/828)* status: 1 of 5 files reviewed, 4 unresolved discussions (waiting on @pkaminski)

*[a discussion](https://reviewable.io/reviews/reviewable/reviewable/828#-MYB0Y5gBWv0VBrn9Yvs:-MYBIc-h7TDbQlQ-oI-f:bu6quqc) (no related file):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

Hmm, indeed, I didn't expect to see Anthony's commit in your PR.  Wanna try to figure it out?  If you get stuck, I'll see if I can stage a rescue tomorrow.  :)
</blockquote></details>

Think we should be good now. 

---
___
*[docs/files.md, line 16 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS4-r2-16:-MYBHkFh9EGS6MaXGXEd:b-akjd85) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/docs/files.md#L16)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

I don't think you addressed the accuracy part of the comment?
</blockquote></details>

I took out `final`, but I assume you mean the part about the icon?  Done.
___
*[docs/files.md, line 35 at r3](https://reviewable.io/reviews/reviewable/reviewable/828#-MYBDXnh3QAHgGEOAaJd:-MYBJj6X4xfhxJBuspoc:b4uhqkx) ([raw file](https://github.com/reviewable/reviewable/blob/587f5d07d7b658e3fe503514514e05eb3e311baf/docs/files.md#L35)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

This anchor is also unnecessary.
</blockquote></details>

I knew I took that out, it was the last thing I did... before not saving the file!
___
*[docs/images/filematrix_12.png, line  at r3](https://reviewable.io/reviews/reviewable/reviewable/828#-MYBERoxEFL0br9XZ_gy:-MYBa61GB6S9_tCcsqwz:b-896fix) ([raw file](https://github.com/reviewable/reviewable/blob/587f5d07d7b658e3fe503514514e05eb3e311baf/docs/images/filematrix_12.png#L)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

File no longer used, revert.
</blockquote></details>

Done.
___
*[enterprise/changelog.md, line 15 at r2](https://reviewable.io/reviews/reviewable/reviewable/828#-MY6K4aEQCb1-eXyibS8-r2-15:-MYBHPkl3Bw0BlAjPPJV:bh8wqg2) ([raw file](https://github.com/reviewable/reviewable/blob/3793dc3688d54f094ca55b5733621d56b343b0ce/enterprise/changelog.md#L15)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

...and that entry about the reverted marker is also just an `upd`.  :)  Basically, if it just adds on to an existing feature (e.g., the file matrix) I treat it as an update unless it's really big.
</blockquote></details>

Heh, I went back and forth on that.  That makes it more clear, thanks.

**Review by pkaminski — APPROVED** [review]
> Reviewed 4 of 5 files at r4.
> *[Reviewable](https://reviewable.io/reviews/reviewable/reviewable/828)* status: all files reviewed, 1 unresolved discussion (waiting on @cgiroux86)
